src/components/number_input.tsx

```diff
@@ -38,21 +39,15 @@ const NumberInput: FC<NumberInputProps> = ({
     value,
     min,
     max,
+    hint,
 }): any => {
     const [error, setError] = useState<string>('')
-
-    // use the cursor postion when user edits the data in the component
-    const [cursor, setCursor] = useState<number | null>(null)
-    const ref = useRef<HTMLInputElement>(null)
-
-    // Refresh after the first render and every time the component updates
-    useEffect(() => {
-        const input = ref.current
-        if (input) input.setSelectionRange(cursor, cursor)
-    }, [ref, cursor, value])
+    const [localValue, setLocalValue] = useState<string>(
+        value as unknown as string,
+    )
 
     const handleChange = (inputValue: string): any => {
-        const inputValueNum: number = parseInt(inputValue)
+        const inputValueNum: number = parseFloat(inputValue)
```

Comment:
We should send this inputValueNum to `updateValue`. In other words, we should be storing the number as a number in the document.

Reply:
updateValue in number_input_wrapper.tsx uses parseFloat to upsertData which store value as float in the database. Changing updateValue with InputValueNum for some reason not working when I tested.

Reply:
Ah. Ok. I didn't notice that value was being converted to a float in number_input_wrapper. That serves the same purpose.  I am resolved on this.